SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -559,8 +592,9 @@ void UGlobalStateManager::TriggerBeginPlay()
 }
 
 // Queries for the GlobalStateManager in the deployment.
-// bRetryUntilAcceptingPlayers will continue querying until the state of AcceptingPlayers is true, this is so clients know when to connect to the deployment.
-void UGlobalStateManager::QueryGSM(bool bRetryUntilAcceptingPlayers)
+// bRetryUntilAcceptingPlayers will continue querying until the state of AcceptingPlayers and SessionId are the same as the given arguments
+// This is so clients know when to connect to the deployment.
+void UGlobalStateManager::QueryGSM(bool AcceptingPlayersToCheck, int32 SessionIdToCheck, const QuerySuccessDelegate& Callback, bool bRetryUntilAcceptingPlayers /*=true*/)
```

Comment:
```suggestion
void UGlobalStateManager::QueryGSM(bool bExpectedAcceptingPlayers, int32 SessionIdToCheck, const QuerySuccessDelegate& Callback, bool bRetryUntilAcceptingPlayers /*=true*/)
```
Let's also comment on what this bool is used for.

Reply:
Also `bRetryUntilAcceptingPlayers` feels like it doesn't do what it used to since it existed to make sure the query would repeat until accepting players was true. 

Also: Is there a case where we would want `AcceptingPlayerstoCheck`/ `bExpectedAcceptingPlayers` to be false?

Reply:
"Also: Is there a case where we would want AcceptingPlayerstoCheck/ bExpectedAcceptingPlayers to be false?" No, at the moment we only check for acceptingPlayers true. 

bRetryUntilAcceptingPlayers -->bRetryUntilRecieveExpectedValues